### DISCUSSION TITLE
Multiple workspace folders

### DISCUSSION THREAD

**Discussion by axieax** [mfussenegger/nvim-jdtls#446] [Q&A]
Hi, I'm currently working on a project with a mono-repo structure, i.e. a repository with multiple Java subdirectory projects (see below). When working on a file in [java-1], I would like jdtls to resolve imports from [java-2], [java-3] and so on.. perhaps using the `vim.lsp.buf.add_workspace_folder` function, although that doesn't seem to work. Is this feature supported, or is there a way to make this work otherwise? Thanks!

```
[repo-root]
- [java-1]
- [java-2]
- [java-3]
- ...
```

**Comment by mfussenegger** [comment]
I think it should work if you set the `workspace_folders` when calling `jdtls.start_or_attach(config)`. But I haven't tried it myself and don't know for sure if eclipse.jdt.ls supports it.


```lua
local config = {
  root_dir = "/path/to/repo_root/",
  workspace_folders = {
    {
      uri = "file:///path/to/repo_root/java_1",
      name = "java_1",
    },
    {
      uri = "file:///path/to/repo_root/java_2",
      name = "java_2",
    },
    {
      uri = "file:///path/to/repo_root/java_3",
      name = "java_3",
    }
  },
}
```

**Reply by axieax** [reply]
Hmm unfortunately I don't think that works either (root_dir is meant to be "/path/to/repo_root/java_1" or jdtls won't think that it is in a project). Perhaps it's easier if there is a way to integrate Bazel into this plugin, do you know if that's possible? It seems like Bazel support has been added to [Eclipse jdtls](https://github.com/eclipse/eclipse.jdt.ls/issues/543), but am not sure how easy it will be to support that with nvim-jdtls.

It may require something like https://github.com/salesforce/bazel-vscode, which I found from https://youtu.be/MVk_zQ_spkc and https://youtu.be/oLnfv2-aGwk
